### DISCUSSION TITLE
What's the correct way to use shallow comparison?

### DISCUSSION THREAD

**Discussion by vincent822** [pmndrs/zustand#2201] [Q&A]
I saw all kind of different suggestions about how to do it (shallow, createWithEqualityFn, etc). A clear explanation would be appreciated.

**Comment by dai-shi** [comment]
https://github.com/pmndrs/zustand/blob/9857a676a94c81ec561f99a13d46d6f6f3376f84/docs/guides/prevent-rerenders-with-use-shallow.md?plain=1#L2 is the current recommendation.

`createWithEqualityFn` and `shallow` is also 100% supported usage (both in v4 and planned v5).
